src/app/controllers/editors/dictionary-preview/dictionary-preview.component.ts

```diff
@@ -0,0 +1,342 @@
+import { HttpErrorResponse } from '@angular/common/http';
+import { Component, Input, OnDestroy, OnInit } from '@angular/core';
+import { TreeNode } from 'primeng/api';
+import { catchError, of, Subject, take } from 'rxjs';
+import { DictionaryNoteVocabo } from 'src/app/models/custom-models/dictionary-note-vocabo';
+import { DictionaryEntry } from 'src/app/models/dictionary/dictionary-entry.model';
+import { DictionaryPreviewItem } from 'src/app/models/dictionary/dictionary-preview-item.model';
+import { DictionarySortingItem } from 'src/app/models/dictionary/dictionary-sorting-item.model';
+import { TextualDocument } from 'src/app/models/dictionary/textual-document.model';
+import { SearchAnnotationFilters, SearchAnnotationRequest } from 'src/app/models/search/search-annotation-request';
+import { SearchAnnotationResult, SearchAnnotationResultRow } from 'src/app/models/search/search-annotation-result';
+import { CommonService } from 'src/app/services/common.service';
+import { DictionaryService } from 'src/app/services/dictionary.service';
+import { LexiconService } from 'src/app/services/lexicon.service';
+import { SearchAnnotationService } from 'src/app/services/search-annotation.service';
+
+/**
+ * classe che rappresenta una accezione (meaning) con le sue annotazioni
+ */
+export class Meaning {
+  id!: string;
+  referredEntity?: string;
+  sortedAnnotations!: SearchAnnotationResult;
+}
+
+/**
+ * classe che rappresenta le accezioni (meanings) di una singola voce (lexical entry del sense)
+ * */
+export class SenseEntry {
+  id!: string;
+  meanings: Meaning[] = [];
+}
+
+@Component({
+  selector: 'app-dictionary-preview',
+  templateUrl: './dictionary-preview.component.html',
+  styleUrls: ['./dictionary-preview.component.scss']
+})
+export class DictionaryPreviewComponent implements OnInit, OnDestroy {
+  private readonly unsubscribe$ = new Subject();
```

Comment:
Al momento non viene usato in alcun punto, a meno che la cosa non cambi è da rimuovere e di conseguenza anche ngOnDestroy